pkg/test/framework/components/istio/istio.go

```diff
@@ -25,6 +27,16 @@ import (
 type Instance interface {
 	resource.Resource
 
+	// Ingress returns a client for interacting with the ingress-gateway in a given cluster. If a nil cluster is provided,
+	// the first configured cluster will be used.
+	Ingress(cluster resource.Cluster) Ingress
```

Comment:
The way this reads, it sounds like the caller needs to know where the ingresses are deployed. It's not necessary to have an ingress in each cluster, and we may eventually define an "ingress topology" that is based around the network/cluster topologies.

So I'm wondering if the spec for this method should be "get me the ingress for workloads residing in this cluster", which may return an ingress in the same cluster or not. The signature may not have to change, however ... just the docs in that case (I think). A subtle change in the name *may* help ... `IngressFor(cluster)`.

Additionally, we could have a method to list all the ingresses ... and each ingress could have the cluster that it's deployed in (in case it's needed).

WDYT?

Reply:
SGTM; I figured this would be the case in some place but couldn't quite find one.